presto-main/src/main/java/io/prestosql/operator/UpdateOperator.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.slice.Slice;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.PageBuilder;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.connector.UpdatablePageSource;
+import io.prestosql.sql.planner.plan.PlanNodeId;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.airlift.concurrent.MoreFutures.toListenableFuture;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.VarbinaryType.VARBINARY;
+import static java.util.Objects.requireNonNull;
+
+public class UpdateOperator
```

Comment:
Apart from the addInput method, this class seems to be duplicating most of the stuff from DeleteOperator class. Should we have an abstract parent?

Reply:
It's a good point.  In general, as you've probably observed, class inheritance is used sparingly in Presto, but this seems a good candidate.

I'll chat @electrum about it.

Reply:
As you observed, nearly all of the code duplicated in DeleteOperator and UpdateOperator can be shared.  I introduced new base class AbstractRowChangingOperator, and changed DeleteOperator and UpdateOperator to inherit from it.

A very nice cleanup!